### PR TITLE
Commit to spring-boot-practice-shopping-cart-branch First

### PR DESCRIPTION
**Pull request by romaxa11**

### CODE REVIEW

File: src/main/java/mate/academy/intro/dto/cartitem/CreateCartItemRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class CreateCartItemRequestDto {
+    @NotNull
+    @Min(1)
+    private Long bookId;
+    @NotNull
+    @Min(0)
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/dto/cartitem/CreateCartItemRequestDto.java]:
```suggestion
    @Min(1)
```

File: src/main/java/mate/academy/intro/model/CartItem.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "shopping_cart_id", nullable = false)
+    private ShoppingCart shoppingCart;
+    @ManyToOne
+    @JoinColumn(name = "book_id", nullable = false)
+    private Book book;
+    @Min(0)
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/model/CartItem.java:35]:
```suggestion
    @Min(1)
```

File: src/main/java/mate/academy/intro/model/ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.Setter;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@Entity
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/model/ShoppingCart.java]:
```suggestion
```

File: src/main/java/mate/academy/intro/model/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    private Long id;
+    @OneToOne(cascade = CascadeType.ALL)
+    @MapsId
+    @PrimaryKeyJoinColumn(name = "user_id")
+    private User user;
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/model/ShoppingCart.java]:
```suggestion
    @OneToOne(cascade = CascadeType.ALL)
    @MapsId
    private User user;
```

File: src/main/java/mate/academy/intro/repository/cartitem/CartItemRepository.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+import mate.academy.intro.model.CartItem;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+
+public interface CartItemRepository extends JpaRepository<CartItem, Long> {
+    @Modifying
+    @Query("DELETE FROM CartItem c WHERE c.shoppingCart.id = :shoppingCartId")
+    void deleteByShoppingCart_Id(Long shoppingCartId);
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/repository/cartitem/CartItemRepository.java:11]: we should use soft delete

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/intro/controller/ShoppingCartController.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    public ShoppingCartResponseDto updateCartItem(
+            Authentication authentication,
+            @PathVariable Long itemId,
+            @RequestBody @Valid UpdateCartItemRequestDto requestDto
+    ) {
+        User user = (User) authentication.getPrincipal();
+        return shoppingCartService.updateCartItem(user.getId(), itemId, requestDto);
+    }
+
+    @DeleteMapping("cart-items/{cartItemId}")
```
Comment (aaananas) [inline src/main/java/mate/academy/intro/controller/ShoppingCartController.java]:
```suggestion
    @DeleteMapping("/cart-items/{cartItemId}")
```

File: src/main/java/mate/academy/intro/controller/ShoppingCartController.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    @PreAuthorize("hasRole('USER')")
+    public ShoppingCartResponseDto addBookToShoppingCart(
+            Authentication authentication,
+            @RequestBody @Valid CreateCartItemRequestDto requestDto
+    ) {
+        User user = (User) authentication.getPrincipal();
+        return shoppingCartService.addCartItem(user.getId(), requestDto);
+    }
+
+    @PutMapping("cart-items/{itemId}")
```
Comment (aaananas) [inline src/main/java/mate/academy/intro/controller/ShoppingCartController.java]:
```suggestion
    @PutMapping("/cart-items/{itemId}")
```

**Review by aaananas — APPROVED** [review]

**Pull request merged**